recipe/schemas.py

```diff
@@ -266,7 +277,282 @@ def _coerce_filter(v):
         },
         'order_by': {
             'type': 'list',
-            'schema': {'type': 'string'},
+            'schema': {
+                'type': 'string'
+            },
         },
     }
 }
+
+format_lookup = {
+    'comma': ',.0f',
+    'dollar': '$,.0f',
+    'percent': '.0%',
+    'comma1': ',.1f',
+    'dollar1': '$,.1f',
+    'percent1': '.1%',
+    'comma2': ',.2f',
+    'dollar2': '$,.2f',
+    'percent2': '.2%',
+}
+
+default_aggregation = 'sum'
+no_aggregation = 'none'
+aggregations = {
+    'sum': func.sum,
+    'min': func.min,
+    'max': func.max,
+    'avg': func.avg,
+    'count': func.count,
+    'count_distinct': lambda fld: func.count(distinct(fld)),
+    'month': lambda fld: func.date_trunc('month', fld),
+    'week': lambda fld: func.date_trunc('week', fld),
+    'year': lambda fld: func.date_trunc('year', fld),
+    'quarter': lambda fld: func.date_trunc('quarter', fld),
+    'age': lambda fld: func.date_part('year', func.age(fld)),
+    'none': lambda fld: fld,
+    None: lambda fld: fld,
+}
+
+aggr_keys = '|'.join(
+    k for k in aggregations.keys() if isinstance(k, basestring)
+)
+# Match patterns like sum(a)
+field_pattern = re.compile(r'^({})\((.*)\)$'.format(aggr_keys))
+
+
+def find_operators(value):
+    """ Find operators in a field that may look like "a+b-c" """
+    field = re.split('[+-/*]', value)[0]
+
+    operator_lookup = {'-': 'sub', '+': 'add', '/': 'div', '*': 'mul'}
+
+    operators = []
+    for part in re.findall('[+-/*]\w+', value):
+        # TODO: Full validation on other fields
```

Comment:
Would the 'other fields' referenced in this TODO be `value`, `aggregation`, `condition`, and `format`?

Reply:
@stevenwally I think what I meant here is right now we only support `sum(a+b+c)`. With more regex we could support `sum(min(a) - max(b))`. I'm not sure that's valid sql though.